src/models/ITestType.ts

```diff
@@ -37,16 +65,24 @@ export interface NextTestTypesOrCategory {
     forVehicleSize: ForVehicleSize[] | null;
     forVehicleConfiguration: ForVehicleConfiguration[] | null;
     forVehicleAxles: number[] | null;
+    forVehicleCategory: ForVehicleCategory[] | null;
+    forVehicleClass: number[] | null;
+    forVehicleSubclass: ForVehicleSubclass[] | null;
+    forVehicleWheels: number[] | null;
     testTypeClassification?: TestTypeClassification;
     testCodes?: TestCode[];
     nextTestTypesOrCategories?: NextTestTypesOrCategory[];
 }
 
 export interface TestCode {
-    forVehicleType: ForVehicleType;
+    forVehicleType: ForVehicleType | ForVehicleType[];
```

Comment:
Should it not be one or the other?

Reply:
Someone had to endure this pain already:
https://github.com/dvsa/cvs-svc-test-types/blob/957b68037a4d296132b0814f1caf4a16a34468ae/src/services/TestTypesService.ts#L164

Reply:
Precisely why I want to avoid it 😜

Reply:
yes, i already had this discussion with Bogdan Pecete. i understood that they should all have been arrays in the first place. but because this is how the taxonomy was developed months ago, i didn't want to break something. i've added _ForVehicleType[]_ because since specialist tests, there are test types for which will exist only one test code regardless the vehicle type.

Reply:
Not sticking everything in an array is going to bite us in the arse, but if Bogdan is happy to sign it off.... :/